@here/harp-utils/lib/UrlResolver.ts

```diff
@@ -4,38 +4,45 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { getAppBaseUrl } from "./UrlPlatformUtils";
+import { resolveReferenceUrl } from "./UrlUtils";
+
 /**
  * Convenience type definition for URL resolvers.
  */
 export type UrlResolver = (url: string) => string;
 
-/**
- * Resolves URL using default URL resolver.
- *
- * By default URL resolver is just identity function, it can be changed using
- * [[setDefaultUrlResolver].
- */
-export function defaultUrlResolver(url: string): string {
-    if (customDefaultUrlResolver !== undefined) {
-        return customDefaultUrlResolver(url);
-    } else {
-        return url;
-    }
+export interface SimpleUrlResolverDefinition {
+    [prefix: string]: string;
 }
 
 /**
- * Change resolver used by [[defaultUrlResolver]].
- *
- * `undefined` resets default resolver to identity function.
+ * Create simple `UrlResolver` that matches `URI` by prefixes and just
+ * rewrites whole dictionary `key` to `value` under key.
  *
- * @param resolver
+ * Example:
+ *    # create resolver that will
+ *    const urlResolver = createSimpleUrlResolver({
+ *       # load font-catalog from CDN
+ *       "npm://@here/harp-fontcatalog/":
+ *            "https://unpkg.com/@here/harp-fontcatalog/",
+ *       # but theme theme assets from same app base url
+ *       "npm://@here/harp-map-theme/resources": "/resources/"
+ *    })
  */
-export function setDefaultUrlResolver(resolver: UrlResolver | undefined) {
-    customDefaultUrlResolver = resolver;
+export function createSimpleUrlResolver(definition: SimpleUrlResolverDefinition) {
+    return (uri: string) => {
```

Comment:
I would prefer to have a class for the UrlResolver

Reply:
done